src/db/deposits.ts

```diff
@@ -9,11 +9,55 @@ import { hidePassword } from "@/utils/auth";
 import CONFIG from "@/config";
 import { RoledPlayer } from "@/types/response/players";
 import { ERR } from "@/config/errors";
-import { CustomError } from "@/middlewares/errorHandler";
+import { CustomError } from "@/helpers/error/CustomError";
+import { OrderBy } from "@/types/request/players";
 
 const prisma = new PrismaClient();
 
 export class DepositsDAO {
+  static _getAll = async (
+    page: number,
+    itemsPerPage: number,
+    search?: string,
+    orderBy?: OrderBy<Deposit>,
+  ): Promise<Deposit[]> => {
+    try {
+      const deposits = await prisma.deposit.findMany({
+        skip: page * itemsPerPage,
+        take: itemsPerPage,
```

Comment:
**suggestion (performance):** Good addition of paginated and searchable queries

The new _getAll method with pagination and search functionality is a good addition for handling large datasets efficiently. Consider adding indexes to the database for the fields used in the search to optimize query performance.